03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+landmasses; let's make the plot a little bit clearer by keeping only the largest 12 landmasses. We do this using the `top_n` function.
+Then to help us make sure the labels have enough space, we'll use horizontal bars instead of vertical ones. We do this using the 
+`coord_flip` function, which swaps the `x` and `y` coordinate axes:
+
+```{r 03-data-islands-bar-2}
+islands_top12 <- top_n(islands_df, 12, size)
+islands_bar <- ggplot(islands_top12, aes(x = landmass, y=size)) + geom_bar(stat="identity") + coord_flip()
+
+```
+
+```{r 03-data-islands-bar-2-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(islands_bar)
+```
+
+This is definitely clearer now, and allows us to answer our question ("are the top 7 largest landmasses continents?") in
+the affirmative. But the question could be made clearer from the plot by organizing the bars not by alphabetical order
+but by size, and to colour them based on whether or not they are a continent. In order to do this, we 
+use `mutate` to add a column to the data regarding whether or not the landmass is a continent:
+
+```{r 03-data-islands-bar-3}
+islands_top12 <- top_n(islands_df, 12, size)
+continents <- c('Africa', 'Antarctica', 'Asia', 'Australia', 'Europe', 'North America', 'South America')
+islands_ct <- mutate(islands_top12, is_continent = ifelse(landmass %in% continents, 'Continent', 'Other'))
+head(islands_ct)
+```
+
+In order to colour the bars, we add the `fill` argument to the aesthetic mapping. Then we use the `reorder` 
+function in the aesthetic mapping to organize the landmasses by their `size` variable.
+Finally, we use the `labs` and `theme` functions to add labels, change the font size, and position the legend:
+
+```{r 03-data-islands-bar-4}
+islands_bar <- ggplot(islands_ct, aes(x = reorder(landmass, size), y = size, fill = is_continent)) + 
+                   geom_bar(stat="identity") +
+                   labs(x = 'Landmass', y = 'Size (1000 square mi)', fill = 'Type') +
+                   coord_flip() +
+                   theme(text = element_text(size = 18), legend.position = c(0.75, 0.45))
+```
+
+```{r 03-data-islands-bar-4-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(islands_bar)
+```
+
+This is now a very effective visualization for answering our original questions. Landmasses are organized by their size,
+and continents are coloured differently than other landmasses, making it quite clear that continents are the largest 7 landmasses.
+
+
+### The Old Faithful eruption / waiting time dataset
+This dataset contains measurements of the waiting time between eruptions and the subsequent eruption duration (in minutes). **Question:** Is there a 
+relationship between the waiting time before an eruption to the duration of the eruption? 
+
+```{r 03-data-faithful}
+# old faithful eruption time / wait time data
```

Comment:
Should we make a note that this is a built-in R data set and thus that is why we don't have to read it in? We probably should do that anytime that's the case.